bridge/irc_listener.go

```diff
@@ -64,8 +113,11 @@ func (i *ircListener) nickTrackQuit(e *irc.Event) {
 func (i *ircListener) OnJoinQuitSettingChange() {
 	// Clear Nicktrack QUIT callback as it races with this
 	i.ClearCallback("QUIT")
-	i.ClearCallback("NICK")
-	i.AddCallback("NICK", i.nickTrackNick)
+
+	if i.relayNickTrackID != 0 {
```

Comment:
Instead of a separate field for this, I think we could move https://github.com/qaisjp/go-discord-irc/blob/f7f3a4822d896bb295609efe7211ade78074d84e/bridge/irc_listener.go#L83-L86 out of the if branch, and rename `joinQuitCallbacks` to something else

Reply:
Check the changes, I think the most recent commit does what you're looking for.

Reply:
The other reason to leave in a `nickTrackQuit` and `nickTrackNick` vs merging them with other functions is to insure that we never could ever have event callback id 0. Worst case scenario the aforementioned ones would have that id, not the ones that are being removed/added in.